Designed to allow setting of string parameters via slave option or menu.
Could possibly be extended to other options with a variable third argument
giving the parameter type, and the associated conversion function. Does
not change the base config file- this should be done externally to MPlayer.
diff -r 1abf56096b14 input/input.c
--- a/input/input.c	Sun Apr 30 16:54:45 2006 +0200
+++ b/input/input.c	Sun Apr 30 16:54:45 2006 +0200
@@ -86,6 +86,7 @@ static mp_cmd_t mp_cmds[] = {
   { MP_CMD_GET_TIME_POS, "get_time_pos", 0, { {-1,{0}} } },
   { MP_CMD_GET_TIME_LENGTH, "get_time_length", 0, { {-1,{0}} } },
   { MP_CMD_SWITCH_AUDIO, "switch_audio", 0, { { MP_CMD_ARG_INT,{-1} }, {-1,{0}} } },
+  { MP_CMD_SET_OPTION, "set_option", 2, { { MP_CMD_ARG_STRING, {0}}, { MP_CMD_ARG_STRING, {0}}, {-1,{0}} } },
 #ifdef USE_TV
   { MP_CMD_TV_STEP_CHANNEL, "tv_step_channel", 1,  { { MP_CMD_ARG_INT ,{0}}, {-1,{0}} }},
   { MP_CMD_TV_STEP_NORM, "tv_step_norm",0, { {-1,{0}} }  },
--- a/input/input.h	2007-06-02 12:54:50.000000000 +0800
+++ b/input/input.h	2007-06-02 12:52:53.000000000 +0800
@@ -93,6 +93,7 @@
 #define MP_CMD_STEP_PROPERTY 91
 #define MP_CMD_RADIO_STEP_FREQ 92
 #define MP_CMD_TV_STEP_FREQ 93
+#define MP_CMD_SET_OPTION 94
 
 #define MP_CMD_GUI_EVENTS       5000
 #define MP_CMD_GUI_LOADFILE     5001
--- a/command.c	2007-06-02 12:54:50.000000000 +0800
+++ b/command.c	2007-06-02 12:51:39.000000000 +0800
@@ -1810,6 +1809,12 @@
 	    exit_player_with_rc(MSGTR_Exit_quit,
 				(cmd->nargs > 0) ? cmd->args[0].v.i : 0);
 
+    case MP_CMD_SET_OPTION: {
+        extern mconfig;
+        char *option = cmd->args[0].v.s, *value = cmd->args[1].v.s;
+        m_config_set_option(mconfig, option, value);
+    } break;
+
 	case MP_CMD_PLAY_TREE_STEP:{
 		int n = cmd->args[0].v.i == 0 ? 1 : cmd->args[0].v.i;
 		int force = cmd->args[1].v.i;
